doc/Route_priority_and_TTL.md

```diff
@@ -0,0 +1,131 @@
+# Route priority and time-to-live
+
+## Context
+In environments where the upstream connection has limited throughput (e.g. cellular or satellite), often times certain critical data needs to take precedence over the mass of normal telemetry messages to ensure speedy delivery. Prioritized messages and TTL addresses this concern by letting the operator specify which routes will be processed first, and how long messages should be retained when offline.
+
+## Glossary
+|Term       | Description   |
+|-----------|---------------|
+|*Routes*   | Declarations used to specify how messages should be passed between modules, IoT Hub, and any leaf devices.|
+|*TTL*      | Time-To-Live, used to specify how long before unsent messages expire in storage.|
+|*FSM*      | Finite state machine|
+
+
+## Design details
+Message priorities will be a new property of message routes.  This is because, when given a set of messages, both the set definition and the endpoint are necessary for the priority to be meaningful, which naturally lends to associating priorities with routes.
```

Comment:
nitpick: should not we use the present tense instead of the future tense in our docs?

Reply:
yea, I updated some of them when I was moving the text from the original dev spec, but it looks like I missed some others.